src/betterproto/templates/template.py.j2

```diff
@@ -154,6 +155,92 @@ class {{ service.py_name }}Stub(betterproto.ServiceStub):
     {% endfor %}
 {% endfor %}
 
+{% for service in output_file.services %}
+class {{ service.py_name }}Base(ServiceBase):
+    {% if service.comment %}
+{{ service.comment }}
+
+    {% endif %}
+
+    {% for method in service.methods %}
+    async def {{ method.py_name }}(self
+        {%- if not method.client_streaming -%}
+            {%- if method.py_input_message and method.py_input_message.fields -%},
+                {%- for field in method.py_input_message.fields -%}
+                    {{ field.py_name }}: {% if field.py_name in method.mutable_default_args and not field.annotation.startswith("Optional[") -%}
+                                            Optional[{{ field.annotation }}]
+                                         {%- else -%}
+                                            {{ field.annotation }}
+                                         {%- endif -%}
+                    {%- if not loop.last %}, {% endif -%}
+                {%- endfor -%}
+            {%- endif -%}
+        {%- else -%}
+            {# Client streaming: need a request iterator instead #}
+            , {{ method.py_input_message.py_name_as_field }}_iterator: AsyncIterable["{{ method.py_input_message_type }}"]
```

Comment:
what about just calling the argument request_iterator? I figure consistency is worth something here and the type annotation is already more descriptive.

Reply:
Also is the type not more specifically an AsyncIterator?

Reply:
> what about just calling the argument request_stream?
It's called `request_iterator` on the client-facing side. I suggest that we name it that if we want more consistency.

Reply:
> Also is the type not more specifically an AsyncIterator?

You're right, I'm specifically calling `stream.__aiter__`. I'll change this.